core/src/main/scala/kafka/server/BrokerToControllerChannelManager.scala

```diff
@@ -364,11 +399,13 @@ class BrokerToControllerRequestThread(
   }
 
   override def doWork(): Unit = {
```

Comment:
It is still the same code. Instead of calling in the `else` case, I'm just calling as soon as I get into the function. As long as the controller type is not changing, the function `maybeResetNetworkClient` does nothing.

Reply:
As discussed offline, by proactively checking if the controller has changed (L402), we avoid falling into the case where an authentication of version mismatch error prevents the channel manager from finding a new controller.